cdap-ui/app/cdap/components/FastAction/SetPreferenceAction/SetPreferenceModal.js

```diff
@@ -0,0 +1,453 @@
+            resolvedPrefArray = orderBy(resolvedPrefArray, [this.state.sortByAttribute], [this.state.sortOrder]);
+            this.setState({
+             inheritedPreferences: resolvedPrefArray
+            });
+          },
+          (error) => {
+            this.setState({
+             error: isObject(error) ? error.response : error
+            });
+          }
+        )
+    );
+  }
+
+  setPreferences() {
+    this.setState({saving: true});
+    this.apiSubscriptions.push(
+      this.setPreferencesApi(this.params, this.getKeyValObject())
+        .subscribe(
+          () => {
+            if (this.props.onSuccess && this.props.setAtNamespaceLevel) {
+              this.props.onSuccess();
+            }
+            this.props.toggleModal();
+          },
+          (error) => {
+            this.setState({
+              error: isObject(error) ? error.response : error
+            });
+          }
+        )
+    );
+  }
+
+  getKeyValPair(prefObj) {
+    let prefArray = [];
+    for (let key in prefObj) {
+      if (prefObj.hasOwnProperty(key)) {
+        prefArray.push({
+          key: key,
+          value: prefObj[key],
+          uniqueId: shortid.generate()
+        });
+      }
+    }
+    return prefArray;
+  }
+
+  getKeyValObject() {
+    let keyValArr = this.state.keyValues.pairs;
+    let keyValObj = {};
+    keyValArr.forEach((pair) => {
+      if (pair.key.length > 0 && pair.value.length > 0) {
+        keyValObj[pair.key] = pair.value;
+      }
+    });
+    return keyValObj;
+  }
+
+  toggleSorted(attribute) {
+    let sortOrder = 'asc';
+    if (this.state.sortByAttribute != attribute) {
+      this.setState({sortOrder});
+    } else {
+      if (this.state.sortOrder === 'asc') {
+        sortOrder = 'desc';
+      }
+      this.setState({sortOrder});
+    }
+    this.setState({sortByAttribute: attribute});
+    let newInheritedPreferences = orderBy(this.state.inheritedPreferences, attribute, sortOrder);
+    this.setState({inheritedPreferences: newInheritedPreferences});
+  }
+
+  onKeyValueChange(keyValues) {
+    if (!this.state.fieldsResetted) {
+      this.setState({keyValues});
+    }
+    else {
+      this.setState({fieldsResetted: false});
+    }
+  }
+
+  allFieldsFilled() {
+    if (this.state.keyValues.pairs) {
+      return this.state.keyValues.pairs.every((keyValuePair) => {
+        return (keyValuePair.key.length > 0 && keyValuePair.value.length > 0);
+      });
+    }
+    return true;
+  }
+
+  resetFields(event) {
+    event.preventDefault();
+    this.getSpecifiedPreferences();
+  }
+
+  preventPropagation(event) {
+    event.stopPropagation();
+    event.nativeEvent.stopImmediatePropagation();
+  }
+
+  componentWillMount() {
+    this.getSpecifiedPreferences();
+    this.getInheritedPreferences();
+  }
+
+  componentWillUnmount() {
+    this.subscription();
+    this.apiSubscriptions.forEach(apiSubscription => apiSubscription.dispose());
+  }
```

Comment:
Oh I wasn't referring to the ordering of the React functions, but to the rules in your link. I think most of those will be enabled if we do [`plugin:react/all`](https://github.com/yannickcr/eslint-plugin-react) in the eslintsrc.json, instead of `plugin:react/recommended` like now.